

On Mon, Sep 13, 2004 at 07:28:27PM -0700, William Lee Irwin III wrote:
> I was informed that the vendor component of the copyright can't be
> clobbered without more care, so this patch retains the older vendor,
> updating it only to reflect the appropriate time period.

/proc/ breaks when PID_MAX_LIMIT is elevated on 32-bit, so this patch
lowers it there. Compiletested on x86-64.

Index: mm5-2.6.9-rc1/include/linux/threads.h
===================================================================

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/include/linux/threads.h |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN include/linux/threads.h~pidhashing-lower-pid_max_limit-for-32-bit-machines include/linux/threads.h
--- 25/include/linux/threads.h~pidhashing-lower-pid_max_limit-for-32-bit-machines	Tue Sep 14 16:41:07 2004
+++ 25-akpm/include/linux/threads.h	Tue Sep 14 16:41:44 2004
@@ -30,6 +30,6 @@
 /*
  * A maximum of 4 million PIDs should be enough for a while:
  */
-#define PID_MAX_LIMIT (4*1024*1024)
+#define PID_MAX_LIMIT (sizeof(long) > 4 ? 4*1024*1024 : PID_MAX_DEFAULT)
 
 #endif
_
